hack/dind-cluster.sh

```diff
@@ -71,175 +57,74 @@ set -o errexit
-}
-
-function docker-exec-script() {
-    local cid=$1
-    local cmd=$2
-
-    ${DOCKER_CMD} exec -t "${cid}" ${BASH_CMD} ${cmd}
-}
+source "$(dirname "${BASH_SOURCE}")/lib/init.sh"
 
 function start() {
+  local origin_root=$1
+  local config_root=$2
+  local deployed_config_root=$3
+  local cluster_id=$4
+  local network_plugin=$5
+  local wait_for_cluster=$6
+  local node_count=$7
+
   # docker-in-docker's use of volumes is not compatible with SELinux
   check-selinux
 
-  echo "Configured network plugin: ${NETWORK_PLUGIN}"
-
-  # TODO(marun) - perform these operations in a container for boot2docker compat
-  echo "Ensuring compatible host configuration"
-  sudo modprobe openvswitch
-  sudo modprobe br_netfilter 2> /dev/null || true
-  sudo sysctl -w net.bridge.bridge-nf-call-iptables=0 > /dev/null
-  # overlayfs, if available, will be faster than vfs
-  sudo modprobe overlay 2> /dev/null || true
-  mkdir -p "${CONFIG_ROOT}"
-
-  if [[ "${SKIP_BUILD}" = "true" ]]; then
-    echo "WARNING: Skipping image build due to OPENSHIFT_SKIP_BUILD=true"
-  else
-    build-images
-  fi
+  echo "Starting dind cluster '${cluster_id}' with plugin '${network_plugin}'"
 
-  ## Create containers
-  echo "Launching containers"
-  local root_volume="-v ${OS_ROOT}:${DEPLOYED_ROOT}"
-  local config_volume="-v ${CONFIG_ROOT}:${DEPLOYED_CONFIG_ROOT}"
-  local volumes="${root_volume} ${config_volume}"
-  # systemd requires RTMIN+3 to shutdown properly
-  local stop="--stop-signal=$(kill -l RTMIN+3)"
-  local base_run_cmd="${DOCKER_CMD} run -dt ${stop} ${volumes}"
-
-  local master_cid="$(${base_run_cmd} --privileged --name="${MASTER_NAME}" \
-      --hostname="${MASTER_NAME}" "${DIND_IMAGE}")"
-  local master_ip="$(get-docker-ip "${master_cid}")"
-
-  local node_cids=()
-  local node_ips=()
+  # Ensuring compatible host configuration
+  #
+  # Running in a container ensures that the docker host will be affected even
+  # if docker is running remotely.  The openshift/dind-node image was chosen
+  # due to its having sysctl installed.
+  ${DOCKER_CMD} run --privileged --net=host --rm -v /lib/modules:/lib/modules \
+                openshift/dind-node bash -e -c \
+                '/usr/sbin/sysctl -w net.bridge.bridge-nf-call-iptables=0 > /dev/null;
+                modprobe openvswitch;
+                modprobe overlay 2> /dev/null || true;
+                modprobe br_netfilter 2> /dev/null || true;'
+
+  # Initialize the cluster config path
+  mkdir -p "${config_root}"
+  echo "${network_plugin}" > "${config_root}/network-plugin"
+  copy-runtime "${origin_root}" "${config_root}/"
+
+  local volumes="-v ${config_root}:${deployed_config_root}"
+  local run_cmd="${DOCKER_CMD} run -dt ${volumes}  --privileged"
+
+  # Create containers
+  local master_cid
+  master_cid="$(${run_cmd} --name="${MASTER_NAME}" --hostname="${MASTER_NAME}" "${MASTER_IMAGE}")"
+  local master_ip
+  master_ip="$(get-docker-ip "${master_cid}")"
```

Comment:
I meant changing the above to:

    local master_cid master_ip
    master_cid="$(${run_cmd} --name="${MASTER_NAME}" --hostname="${MASTER_NAME}" "${MASTER_IMAGE}")"
    master_ip="$(get-docker-ip "${master_cid}")"

ie, declaration assignment assignment, rather than declaration assignment declaration assignment